lineapy/api/api.py

```diff
@@ -136,6 +140,60 @@ def save(reference: object, name: str) -> LineaArtifact:
     return linea_artifact
 
 
+def delete(
+    artifact_name: str, version: Optional[Union[int, str]] = None
+) -> None:
+    """
+    Deletes an artifact from artifact store. If no other artifacts
+    refer to the value, the value is also deleted from both the
+    value node store and the pickle store.
+
+    If version is not provided, latest version is used.
+    """
+    execution_context = get_context()
+    executor = execution_context.executor
+    db = executor.db
+
+    get_version = None if not isinstance(version, int) else version
+    artifact = db.get_artifact_by_name(artifact_name, version=get_version)
+
+    node_id = artifact.node_id
+    execution_id = artifact.execution_id
+
+    num_artifacts = db.number_of_artifacts_per_node(node_id, execution_id)
+    if num_artifacts == 1:
+        try:
+            pickled_path = db.get_node_value_path(node_id, execution_id)
+            db.delete_node_value_from_db(node_id, execution_id)
+            if pickled_path is not None:
+                try:
+                    _try_delete_from_db(Path(pickled_path))
+                except KeyError:
+                    logging.info(f"Pickle not found at {pickled_path}")
+            else:
+                logging.info(f"No pickle associated with {node_id}")
+        except ValueError:
+            logging.info(f"No pickle associated with {node_id}")
+
+    delete_version = version or "latest"
+    db.delete_artifact_by_name(artifact_name, version=delete_version)
+
+
+def _try_delete_from_db(pickled_path: Path) -> None:
+    if pickled_path.exists():
+        pickled_path.unlink()
```

Comment:
TIL https://stackoverflow.com/questions/42636018/python-difference-between-os-remove-and-os-unlink-and-which-one-to-use